bootstring/src/main/scala/cats/bootstring/Base.scala

```diff
@@ -56,62 +98,79 @@ object Base {
-              .catchNonFatal(
-                unsafeDigitToCodePoint(digit, uppercase)
-              )
-              .leftMap(_.getLocalizedMessage)
-
-          final override def codePointToDigit(codePoint: Int): Either[String, Int] =
-            ApplicativeError[Either[Throwable, *], Throwable]
-              .catchNonFatal(
-                unsafeCodePointToDigit(codePoint)
-              )
-              .leftMap(_.getLocalizedMessage)
-        }
-      )
-    } else {
-      Left(
-        s"Base values must be equal to the size of the defined code points for each base digit. codePointList size is ${codePointList.size}, base value is ${baseValue}")
-    }
+    unsafeFrom(
+      36,
+      d => intToCodePointDigit(d, lowercaseArray),
+      d => intToCodePointDigit(d, uppercaseArray),
+      codePointDigitToInt)
+  }
 
+  /**
+   * Create a [[Base]] value.
+   *
+   * @param baseValue
+   *   The number of digits in the base.
+   * @param intToLowerCodePointF
+   *   A function to convert an integer value to a the Unicode code point for the digit with the
+   *   corresponding value in this base with a lower case representation. This may throw on
+   *   invalid input.
+   * @param intToUpperCodePointF
+   *   A function to convert an integer value to a the Unicode code point for the digit with the
+   *   corresponding value in this base with an upper case representation. This may throw on
+   *   invalid input.
+   * @param codePointDigitToIntF
+   *   A function to convert a Unicode code point representing a digit in this base to the
+   *   corresponding numerical value. This may throw on invalid input.
+   */
   def unsafeFrom(
       baseValue: Int,
-      codePointList: List[Char],
-      codePointToDigit: Int => Int
+      intToLowerCodePointF: Int => Int,
+      intToUpperCodePointF: Int => Int,
+      codePointDigitToIntF: Int => Int
   ): Base =
-    from(baseValue, codePointList, codePointToDigit).fold(
-      e => throw new IllegalArgumentException(e),
-      identity
-    )
+    new Base {
+      override val value: Int = baseValue
+
+      final override def unsafeIntToCodePointDigit(digit: Int, uppercase: Boolean): Int =
+        if (uppercase) {
+          intToUpperCodePointF(digit)
+        } else {
+          intToLowerCodePointF(digit)
+        }
+
+      final override def unsafeCodePointDigitToInt(codePoint: Int): Int =
+        codePointDigitToIntF(codePoint)
+
+      final override def intToCodePointDigit(
+          digit: Int,
+          uppercase: Boolean): Either[String, Int] =
+        ApplicativeError[Either[Throwable, *], Throwable]
+          .catchNonFatal(
+            unsafeIntToCodePointDigit(digit, uppercase)
+          )
+          .leftMap(_ => s"Digit $digit is not valid for the given base.")
+
+      final override def codePointDigitToInt(codePoint: Int): Either[String, Int] =
+        ApplicativeError[Either[Throwable, *], Throwable]
+          .catchNonFatal(
```

Comment:
There's a special syntax for this :)

```suggestion
        Either.catchNonFatal(
```

Reply:
Really?! How did I not know about this? Thanks I'll update that. I've always been annoyed by the verbosity of this.